### PR TITLE
CMake: Bump required version

### PR DESCRIPTION
**Pull request by jschueller**
Signed-off-by: Julien Schueller <schueller@phimeca.com>

Bump cmake version to 3.5 to avoid this deprecation message:

```
CMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

It should be old enough:
https://cmake.org/pipermail/cmake/2016-March/062947.html

### Description 
_Add a comprehensive description of proposed changes_

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [x] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

**Review by isaevil — COMMENTED** [review]
I am wondering, do we need to bump version to `CMakeLists.txt` located at `examples/` subdirectory as well?

**Comment by jschueller** [comment]
oh sorry

**Comment by phprus** [comment]
https://cmake.org/cmake/help/v3.25/release/3.4.html:

> CMake no longer links executables with flags to export symbols unless the [ENABLE_EXPORTS](https://cmake.org/cmake/help/v3.25/prop_tgt/ENABLE_EXPORTS.html#prop_tgt:ENABLE_EXPORTS) target property is set. See policy [CMP0065](https://cmake.org/cmake/help/v3.25/policy/CMP0065.html#policy:CMP0065).

**Comment by jschueller** [comment]
ok, I have to set it manually

**Review by ilya-lavrenov — APPROVED** [review]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,7 @@ function(tbb_add_test)
     # Define the target for test
     add_executable(${_tbb_test_TARGET_NAME} ${_tbb_test_SUBDIR}/${_tbb_test_NAME}.cpp)
     target_include_directories(${_tbb_test_TARGET_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/.. ${CMAKE_CURRENT_SOURCE_DIR})
+    set_property(TARGET ${_tbb_test_TARGET_NAME} PROPERTY ENABLE_EXPORTS TRUE)
```
Comment (gruenich) [inline test/CMakeLists.txt:31]: Is this change on purpose? I don't see any connection to bumping the CMake version. If such a connection exist, would be good to add it to the commit message.

Comment (jschueller) [reply test/CMakeLists.txt:33]: done

**Review by isaevil — APPROVED** [review]
LGTM
